src/aosm/azext_aosm/build_processors/nexus_image_processor.py

```diff
@@ -0,0 +1,171 @@
+
+        return AzureOperatorNexusNetworkFunctionImageApplication(
+            name=self.name,
+            depends_on_profile=DependsOnProfile(install_depends_on=[],
+                                                uninstall_depends_on=[], update_depends_on=[]),
+            artifact_profile=self._generate_artifact_profile(),
+            deploy_parameters_mapping_rule_profile=self._generate_mapping_rule_profile(),
+        )
+
+    def generate_resource_element_template(self) -> ResourceElementTemplate:
+        """
+        Generate the resource element template.
+
+        :raises NotImplementedError: NSDs do not support deployment of Nexus images.
+        """
+        raise NotImplementedError("NSDs do not support deployment of Nexus images.")
+
+    def _generate_artifact_profile(self) -> AzureOperatorNexusImageArtifactProfile:
+        """
+        Generate the artifact profile.
+
+        :return: The artifact profile.
+        :rtype: AzureOperatorNexusImageArtifactProfile
+        """
+        logger.debug("Generating artifact profile for Nexus image input.")
+        artifact_profile = ImageArtifactProfile(
+            image_name=self.input_artifact.artifact_name,
+            image_version=self.input_artifact.artifact_version,
+        )
+
+        return AzureOperatorNexusImageArtifactProfile(
+            artifact_store=ReferencedResource(id=""),
+            image_artifact_profile=artifact_profile,
+        )
+
+    def _generate_mapping_rule_profile(
+        self,
+    ) -> AzureOperatorNexusImageDeployMappingRuleProfile:
+        """
+        Generate the mapping rule profile.
+
+        :return: The mapping rule profile.
+        :rtype: AzureOperatorNexusImageDeployMappingRuleProfile
+        """
+        logger.debug("Generating mapping rule profile for Nexus image input.")
+        user_configuration = self.generate_values_mappings(
+            self.input_artifact.get_schema(), self.input_artifact.get_defaults()
+        )
+
+        mapping = ImageMappingRuleProfile(
+            user_configuration=json.dumps(user_configuration),
+        )
+
+        return AzureOperatorNexusImageDeployMappingRuleProfile(
+            application_enablement=ApplicationEnablement.ENABLED,
+            image_mapping_rule_profile=mapping,
+        )
+
+    def generate_parameters_file(self) -> LocalFileBuilder:
+        """ Generate parameters file. """
+        mapping_rule_profile = self._generate_mapping_rule_profile()
+        params = (
+            mapping_rule_profile.image_mapping_rule_profile.user_configuration
+        )
+        logger.info(
+            "Created parameters file for Nexus image."
+        )
+        return LocalFileBuilder(
+            Path(
+                VNF_OUTPUT_FOLDER_FILENAME,
+                NF_DEFINITION_FOLDER_NAME,
+                self.input_artifact.artifact_name + '-' + NEXUS_IMAGE_PARAMETERS_FILENAME,
+            ),
+            json.dumps(json.loads(params), indent=4),
+        )
```

Comment:
There is more duplication between Nexus and VHD processors in the above few methods, but the case for refactoring them feels less obvious. Maybe give them a go, and see if the code looks more maintainable. 

(That really goes for all the deduplication comments in this review - if after removing the duplication you think things look harder to maintain, then it may be better to have duplicated code - but you need to make that case, at least to yourself, and possibly to others :-) )